tt_metal/impl/dispatch/kernels/command_queue_consumer.hpp

```diff
@@ -72,7 +74,14 @@ FORCE_INLINE void write_buffers(
             num_to_write = min(num_pages - id, producer_consumer_transfer_num_pages);
             multicore_cb_wait_front(db_buf_switch, num_to_write);
             uint32_t src_addr = get_read_ptr(db_buf_switch);
-            buffer.noc_async_write_buffer(src_addr, id, num_to_write, 0);
+            if((BufferType)dst_buf_type == BufferType::SYSTEM_MEMORY || !sharded){
```

Comment:
spaces (see other comments).

Additiionally, follow dispatch convention of using `not` keyword instead of `!` and using `or` instead of `||`. If there are cases where you are using `&&`, please replace with keyword `and`